lib/normalizer.c

```diff
@@ -2723,17 +2748,39 @@ grn_nfkc_normalize_unify(grn_ctx *ctx,
     need_swap = GRN_TRUE;
   }
 
+  if (data->options->unify_kana_hyphen) {
+    if (need_swap) {
+      grn_nfkc_normalize_context_swap(ctx, &(data->context), &unify);
+      grn_nfkc_normalize_context_rewind(ctx, &unify);
+    }
+    grn_nfkc_normalize_prolonged_sound_mark_like_data prolonged_sound_mark_like_data;
+    prolonged_sound_mark_like_data.is_prolonged_sound_mark_like_char = grn_nfkc_normalize_is_hyphen;
+    prolonged_sound_mark_like_data.previous_length = 0;
```

Comment:
```suggestion
    grn_nfkc_normalize_prolonged_sound_mark_like_data data;
    data.is_prolonged_sound_mark_like_char = grn_nfkc_normalize_is_hyphen;
    data.previous_length = 0;
```

Reply:
The name `data` is already used as argument, this is why I named this variable `prolonged_sound_mark_like_data`.

```
grn_nfkc_normalize_unify(grn_ctx *ctx,
                         grn_nfkc_normalize_data *data)
```

Reply:
And we already use the `data` variable with `prolonged_sound_mark_like_data`.


```
    grn_nfkc_normalize_prolonged_sound_mark_like_data prolonged_sound_mark_like_data;
    prolonged_sound_mark_like_data.is_target_char = grn_nfkc_normalize_is_hyphen;
    prolonged_sound_mark_like_data.previous_length = 0;
    grn_nfkc_normalize_unify_stateful(ctx,
                                      data,
                                      &unify,
                                      grn_nfkc_normalize_unify_kana_prolonged_sound_mark_like,
                                      &prolonged_sound_mark_like_data,
                                      "[unify][kana-hyphen]");
```

Reply:
I see. How about `stateful_data` or `subdata`?

Reply:
I adopted `subdata`.